public/app/features/explore/Logs.tsx

```diff
@@ -169,6 +169,7 @@ export default class Logs extends PureComponent<Props, State> {
     const { deferLogs, renderAll, showLabels, showLocalTime, showUtc } = this.state;
     const { dedupStrategy } = this.props;
     const hasData = data && data.rows && data.rows.length > 0;
+    const hasLabel = hasData && data.rows.some(row => Object.keys(row.uniqueLabels).length > 0);
```

Comment:
wonder if this can be precomputed before when the logs model is built, that way we do not need to to do this in the render path (and using slow function callback)

Reply:
Done. Downside is that now each log datasource will have to reimplement the logic. But that can be abstracted out later by using a base LogsModel instead of just an interface.